src/open_inwoner/laposta/forms.py

```diff
@@ -38,26 +45,45 @@ def __init__(self, user=None, *args, **kwargs):
     def save(self, request, *args, **kwargs):
         newsletters = self.cleaned_data["newsletters"]
 
-        if laposta_client := create_laposta_client():
-            user_data = UserData(
-                ip=get_client_ip(request)[0],
-                email=request.user.email,
-                source_url=None,
-                custom_fields=None,
-                options=None,
+        client = create_laposta_client()
+        if not client:
+            return
+
+        list_name_mapping = dict(self.fields["newsletters"].choices)
+        user_data = UserData(
+            ip=get_client_ip(request)[0],
+            email=request.user.email,
+            source_url=None,
+            custom_fields=None,
+            options=None,
+        )
+        existing_subscriptions = set(
+            Subscription.objects.filter(user=request.user).values_list(
+                "list_id", flat=True
             )
-            existing_subscriptions = set(
-                Subscription.objects.filter(user=request.user).values_list(
-                    "list_id", flat=True
+        )
+
+        to_create = []
+        for list_id in newsletters:
+            if list_id in existing_subscriptions:
+                continue
+
+            try:
+                member = client.create_subscription(list_id, user_data)
+            except (RequestException, ClientError):
+                logger.exception(
+                    "Something went wrong while trying to create subscription"
```

Comment:
Perhaps catch the errros as `exc` or something and add then add `exc_info=exc` to the call

Reply:
`logging.exception` adds the traceback by default: https://docs.python.org/3/library/logging.html#logging.Logger.exception